components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -143,27 +183,30 @@ abstract class EngineSession(
             const val SAFE_BROWSING_ALL =
                 SAFE_BROWSING_MALWARE + SAFE_BROWSING_UNWANTED + SAFE_BROWSING_HARMFUL + SAFE_BROWSING_PHISHING
 
-            internal const val RECOMMENDED = AD + ANALYTICS + SOCIAL + TEST + SAFE_BROWSING_ALL
+            const val RECOMMENDED = AD + ANALYTICS + SOCIAL + TEST + SAFE_BROWSING_ALL
 
-            internal const val ALL = RECOMMENDED + CRYPTOMINING + FINGERPRINTING + CONTENT
+            const val ALL = RECOMMENDED + CRYPTOMINING + FINGERPRINTING + CONTENT
 
-            fun none() = TrackingProtectionPolicy(NONE)
+            fun none() = TrackingProtectionPolicy(NONE, cookiePolicy = ACCEPT_ALL)
 
             /**
              * Strict policy.
              * Combining the [recommended] categories plus [CRYPTOMINING], [FINGERPRINTING] and [CONTENT].
+             *  With a cookiePolicy of [ACCEPT_NON_TRACKERS].
              * This is the strictest setting and may cause issues on some web sites.
              */
             fun all() = TrackingProtectionPolicyForSessionTypes(ALL)
 
             /**
              * Recommended policy.
              * Combining the [AD], [ANALYTICS], [SOCIAL], [TEST] categories plus [SAFE_BROWSING_ALL].
+             * With a [CookiePolicy] of [ACCEPT_NON_TRACKERS].
              * This is the recommended setting.
              */
             fun recommended() = TrackingProtectionPolicyForSessionTypes(RECOMMENDED)
 
-            fun select(vararg categories: Int) = TrackingProtectionPolicyForSessionTypes(categories.sum())
+            fun select(vararg categories: Int, cookiePolicy: CookiePolicy = ACCEPT_NON_TRACKERS) =
```

Comment:
Oh did you decide not to go with a fluent API? I think this is fine / simple enough for now.

Reply:
Yes, I was playing with the fluent API that you suggested me, and I realized that consumers only care about cookie policies, when they want something different from the defaults, this solution was simple enough to allow them customization without introducing a new API.

Reply:
ok, makes sense!